src/matlab2tikz.m

```diff
@@ -2267,12 +2294,19 @@ function assertRegularAxes(handle)
     % that is set when the color == 'none' (if it is omitted, the property will not
     % be set).
     % TODO: probably this should be integrated with getAndCheckDefault etc.
+    if opts_has(options,property) && isNone(opts_get(options,property))
+        return
+    end
     if ~isNone(color)
         [m2t, xcolor] = getColor(m2t, handle, color, 'patch');
         if ~isempty(xcolor)
             % this may happen when color == 'flat' and CData is Nx3, e.g. in
             % scatter plot or in patches
-            options = opts_add(options, property, xcolor);
+            if isempty(property)
+                options = opts_add(options, xcolor);
+            else
+                options = opts_add(options, property, xcolor);
+            end
```

Comment:
@miscco That introduces `key={k=v,k=v,...}` not `key`. Anyhow, there are sections in this PR where `opts_addSubOpts` could come in handy, sure. I will address those upon merge, that's small stuff to fix (it's more complicated to do it using that function, but it provides consistency in the output). IMHO, it's not something to bother a first-timer with (the same for our test suite).

From within the `setColor` function, there was no mechanism to enforce `key` instead of `key=value` until now.